### DISCUSSION TITLE
do not mutate vuex store state outside mutation handlers WARNING

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#9678] [General - Components / Directives / etc] [answered]
Hy guys ... 
I would like to share this issue:
Even my code is working as I want, im recibing a lot of warnings about:
`[Vue warn]: Error in callback for watcher "function () { return this._data.$$state }": "Error: [vuex] do not mutate vuex store state outside mutation handlers."`
Warnings are triggered when modifing some ui component.
I dont know what is it about or what im doing wrong,  this is what im doing:
I am bringing some data from an api and store it the store using a mutation. (getApidata)
Then with a computed property and watcher i, I bring the data into the account object to create the bindings for the ui components.
and this is the code:
```
<template>
  <q-page padding class="flex flex-center">
          <div class="column no-wrap -center">
            <div class="row q-mt-lg">
              <div class="col">
                <q-input
                  outlined
                  label-color="gray"
                  color="julep1"
                  bg-color="input"
                  :v-model="this.account.personal.firstName"
                  :label="this.account.personal.firstName"
                />
              </div>

              <div class="col q-ml-sm">
                <q-input
                  class=""
                  outlined
                  label-color="gray"
                  color="julep1"
                  bg-color="input"
                  :v-model="this.account.personal.lastName"
                  :label="this.account.personal.lastName"
                />
              </div>
            </div>
            <div class="q-mt-md j-text1">Age</div>
            <q-slider
              label
              :label-value="account.personal.age"
              v-model="account.personal.age"
              :min="10"
              :max="100"
              :step="1"
            />
            <div class="q-mt-md j-text1">
              Marital status
            </div>

            <div class="q-gutter-sm ">
              <q-radio
                v-model="account.personal.maritalStatus"
                val="Single"
                label="Single"
              />
              <q-radio
                v-model="account.personal.maritalStatus"
                val="Divorcied"
                label="divorced"
              />
              <q-radio
                v-model="account.personal.maritalStatus"
                val="Married"
                label="Married"
              />
            </div>
            <div class="row j-text1 text-left">Zip code</div>
            <q-input
              class=""
              outlined
              label-color="gray"
              color="julep1"
              bg-color="input"
              :v-model="this.account.personal.zipCode"
              :label="this.account.personal.zipCode"
            />
          </div>
  </q-page>
</template>

<script>
import { getApidata } from "../services/apiServices";

export default {
  name: "Account",
  mounted: function() {
    getApidata("/myprofile", "julepx/UPDATE_ACCOUNT");
  },
  data() {
    return {
      render: false,
      account: {},
    };
  },

  computed: {
    accountState: {
      get() {
        return this.$store.getters["julepx/accountInfo"];
      }
    }
  },
  watch: {
    accountState(value) {
      this.account = value;
      this.render = true;
    }
  }
};
</script>
```

Any advice to improve my code / fix this warning would be wellcome.
Leandro.

**Comment by turigeza** [comment, marked as answer]
What you are doing is not allowed/supported by vuex. If you are using vuex you have to use mutations to alter the vuex state.
The problem is that v-model is changing the state and not doing that via a store mutation.
So instead of v-model you should use :value and @input 

A basic example 
`
:value="this.account.personal.firstName"
@input="SET_FIRST_NAME"
`
Where SET_FIRST_NAME is your store mutation responsible changing the state.

**Reply by leostereo** [reply]
Thanks @turigeza ; Your solution works ! ; just in case some one needs , there is more documentation here:
[https://vuex.vuejs.org/guide/forms.html#two-way-computed-property](url)
Now im just wondering if this is the correct way to achieve what I want.
Since:
I need to take the vuex state into some component local variables , 
Then I need to modify those variables (it is a CRUD activity) using input components.
Then send back to api.
but ....
I dont need to directly  modify back the vuex state when interacting with input components.
So ... perhaps there is another way / tool to achieve this .... just wondering.
thanks!!
